globals.py

```diff
@@ -7,4 +7,9 @@
 player = pg.sprite.GroupSingle()
 player.add(Player())
 
-map = Map()
\ No newline at end of file
+map = Map()
+
+
+# Every singleton should have a lifecycle hook called on_globals_loaded which fires once every singleton has been initialized
+player.sprite.on_globals_loaded()
+map.on_globals_loaded()
```

Comment:
Let's use this "lifecycle hook" which I have called `on_globals_loaded` (we can rename later if you want). Basically, by convention, every class will have a method with this name. Similar to `__init__` which is inherent, the `on_globals_loaded` function will be called once, but specifically only after ALL singletons have been initialized. This is because sometimes classes need to be able to refer to certain values that live within one another, and this avoids the problem of a circular dependency.